### ISSUE TITLE
Build failing

### ISSUE THREAD

**Issue by 94b1** [monero-project/kovri#403]
**By submitting this issue, I confirm the following:**

- I have read and understood the [contributor guide](https://github.com/monero-project/kovri/blob/master/doc/CONTRIBUTING.md).
- I have checked that the issue I am reporting can be replicated or that the feature I am suggesting is not present.
- I have checked opened or recently closed [pull requests](https://github.com/monero-project/kovri/pulls) for existing solutions/implementations to my issue/suggestion.

*Place an X inside the bracket to confirm*
- [X] I confirm.

---

System: Debian 8.6 (fresh install)

The following errors are occurring when compiling with gcc, but I've installed all the dependencies including clang from ```testing```.  

After I run ```make```, I get the following error.

```
$ make
mkdir -p deps/cpp-netlib/build/
cd deps/cpp-netlib/build/ && cmake  -D CMAKE_BUILD_TYPE=Debug -D CPP-NETLIB_BUILD_TESTS=OFF -D CPP-NETLIB_BUILD_EXAMPLES=OFF ../ && make
-- Boost version: 1.61.0
-- Found the following Boost libraries:
--   system
--   thread
--   chrono
--   date_time
--   atomic
-- Configuring done
-- Generating done
-- Build files have been written to: /home/94b1/files/i2p/kovri/deps/cpp-netlib/build
make[1]: Entering directory '/home/94b1/files/i2p/kovri/deps/cpp-netlib/build'
/usr/bin/cmake -H/home/94b1/files/i2p/kovri/deps/cpp-netlib -B/home/94b1/files/i2p/kovri/deps/cpp-netlib/build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /home/94b1/files/i2p/kovri/deps/cpp-netlib/build/CMakeFiles /home/94b1/files/i2p/kovri/deps/cpp-netlib/build/CMakeFiles/progress.marks
make -f CMakeFiles/Makefile2 all
make[2]: Entering directory '/home/94b1/files/i2p/kovri/deps/cpp-netlib/build'
make -f libs/network/src/CMakeFiles/cppnetlib-uri.dir/build.make libs/network/src/CMakeFiles/cppnetlib-uri.dir/depend
make[3]: Entering directory '/home/94b1/files/i2p/kovri/deps/cpp-netlib/build'
cd /home/94b1/files/i2p/kovri/deps/cpp-netlib/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/94b1/files/i2p/kovri/deps/cpp-netlib /home/94b1/files/i2p/kovri/deps/cpp-netlib/libs/network/src /home/94b1/files/i2p/kovri/deps/cpp-netlib/build /home/94b1/files/i2p/kovri/deps/cpp-netlib/build/libs/network/src /home/94b1/files/i2p/kovri/deps/cpp-netlib/build/libs/network/src/CMakeFiles/cppnetlib-uri.dir/DependInfo.cmake --color=
make[3]: Leaving directory '/home/94b1/files/i2p/kovri/deps/cpp-netlib/build'
make -f libs/network/src/CMakeFiles/cppnetlib-uri.dir/build.make libs/network/src/CMakeFiles/cppnetlib-uri.dir/build
make[3]: Entering directory '/home/94b1/files/i2p/kovri/deps/cpp-netlib/build'
[ 14%] Building CXX object libs/network/src/CMakeFiles/cppnetlib-uri.dir/uri/uri.cpp.o
cd /home/94b1/files/i2p/kovri/deps/cpp-netlib/build/libs/network/src && /usr/bin/c++   -DBOOST_NETWORK_DEBUG -DBOOST_NETWORK_ENABLE_HTTPS -DBOOST_SPIRIT_THREADSAFE -DBOOST_TEST_DYN_LINK -I/home/94b1/files/i2p/kovri/deps/cpp-netlib  -maes -march=native  -Wall -std=c++11 -g   -o CMakeFiles/cppnetlib-uri.dir/uri/uri.cpp.o -c /home/94b1/files/i2p/kovri/deps/cpp-netlib/libs/network/src/uri/uri.cpp
c++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-6/README.Bugs> for instructions.
libs/network/src/CMakeFiles/cppnetlib-uri.dir/build.make:65: recipe for target 'libs/network/src/CMakeFiles/cppnetlib-uri.dir/uri/uri.cpp.o' failed
make[3]: *** [libs/network/src/CMakeFiles/cppnetlib-uri.dir/uri/uri.cpp.o] Error 4
make[3]: Leaving directory '/home/94b1/files/i2p/kovri/deps/cpp-netlib/build'
CMakeFiles/Makefile2:92: recipe for target 'libs/network/src/CMakeFiles/cppnetlib-uri.dir/all' failed
make[2]: *** [libs/network/src/CMakeFiles/cppnetlib-uri.dir/all] Error 2
make[2]: Leaving directory '/home/94b1/files/i2p/kovri/deps/cpp-netlib/build'
Makefile:141: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/94b1/files/i2p/kovri/deps/cpp-netlib/build'
Makefile:101: recipe for target 'dependencies' failed
make: *** [dependencies] Error 2
```

I thought maybe the problem was with the compiler and/or debian's sources, so I tried to build it on Ubuntu 16.04 (installed all the dependencies)

I get the following error in Ubuntu (fresh install):

```
$make
mkdir -p deps/cpp-netlib/build/
cd deps/cpp-netlib/build/ && cmake  -D CMAKE_BUILD_TYPE=Debug -D CPP-NETLIB_BUILD_TESTS=OFF -D CPP-NETLIB_BUILD_EXAMPLES=OFF ../ && make
-- Boost version: 1.58.0
-- Found the following Boost libraries:
--   system
--   thread
--   chrono
--   date_time
--   atomic
-- Configuring done
-- Generating done
-- Build files have been written to: /home/94b1/files/git/kovri/deps/cpp-netlib/build
make[1]: Entering directory '/home/94b1/files/git/kovri/deps/cpp-netlib/build'
/usr/bin/cmake -H/home/94b1/files/git/kovri/deps/cpp-netlib -B/home/94b1/files/git/kovri/deps/cpp-netlib/build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /home/94b1/files/git/kovri/deps/cpp-netlib/build/CMakeFiles /home/94b1/files/git/kovri/deps/cpp-netlib/build/CMakeFiles/progress.marks
make -f CMakeFiles/Makefile2 all
make[2]: Entering directory '/home/94b1/files/git/kovri/deps/cpp-netlib/build'
make -f libs/network/src/CMakeFiles/cppnetlib-uri.dir/build.make libs/network/src/CMakeFiles/cppnetlib-uri.dir/depend
make[3]: Entering directory '/home/94b1/files/git/kovri/deps/cpp-netlib/build'
cd /home/94b1/files/git/kovri/deps/cpp-netlib/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/94b1/files/git/kovri/deps/cpp-netlib /home/94b1/files/git/kovri/deps/cpp-netlib/libs/network/src /home/94b1/files/git/kovri/deps/cpp-netlib/build /home/94b1/files/git/kovri/deps/cpp-netlib/build/libs/network/src /home/94b1/files/git/kovri/deps/cpp-netlib/build/libs/network/src/CMakeFiles/cppnetlib-uri.dir/DependInfo.cmake --color=
make[3]: Leaving directory '/home/94b1/files/git/kovri/deps/cpp-netlib/build'
make -f libs/network/src/CMakeFiles/cppnetlib-uri.dir/build.make libs/network/src/CMakeFiles/cppnetlib-uri.dir/build
make[3]: Entering directory '/home/94b1/files/git/kovri/deps/cpp-netlib/build'
[ 14%] Building CXX object libs/network/src/CMakeFiles/cppnetlib-uri.dir/uri/uri.cpp.o
cd /home/94b1/files/git/kovri/deps/cpp-netlib/build/libs/network/src && /usr/bin/c++   -DBOOST_NETWORK_DEBUG -DBOOST_NETWORK_ENABLE_HTTPS -DBOOST_SPIRIT_THREADSAFE -DBOOST_TEST_DYN_LINK -I/home/94b1/files/git/kovri/deps/cpp-netlib  -maes -march=native  -Wall -ftemplate-depth=256 -std=c++11 -g   -o CMakeFiles/cppnetlib-uri.dir/uri/uri.cpp.o -c /home/94b1/files/git/kovri/deps/cpp-netlib/libs/network/src/uri/uri.cpp
clang: error: unable to execute command: Killed
clang: error: clang frontend command failed due to signal (use -v to see invocation)
clang version 3.8.0-2ubuntu4 (tags/RELEASE_380/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
clang: note: diagnostic msg: PLEASE submit a bug report to http://llvm.org/bugs/ and include the crash backtrace, preprocessed source, and associated run script.
clang: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang: note: diagnostic msg: /tmp/uri-db710a.cpp
clang: note: diagnostic msg: /tmp/uri-db710a.sh
clang: note: diagnostic msg: 

********************
libs/network/src/CMakeFiles/cppnetlib-uri.dir/build.make:65: recipe for target 'libs/network/src/CMakeFiles/cppnetlib-uri.dir/uri/uri.cpp.o' failed
make[3]: *** [libs/network/src/CMakeFiles/cppnetlib-uri.dir/uri/uri.cpp.o] Error 254
make[3]: Leaving directory '/home/94b1/files/git/kovri/deps/cpp-netlib/build'
CMakeFiles/Makefile2:92: recipe for target 'libs/network/src/CMakeFiles/cppnetlib-uri.dir/all' failed
make[2]: *** [libs/network/src/CMakeFiles/cppnetlib-uri.dir/all] Error 2
make[2]: Leaving directory '/home/94b1/files/git/kovri/deps/cpp-netlib/build'
Makefile:141: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/94b1/files/git/kovri/deps/cpp-netlib/build'
Makefile:101: recipe for target 'dependencies' failed
make: *** [dependencies] Error 2
```

Should I include the files /tmp/uri-db710a.cpp and /tmp/uri-db710a.sh  ? They're big (30+ MB)

But I've got no build errors on my daily system using clang 3.8 (which is not on a cloud instance like the above builds). Uploading the kovri binary from my system and running it on the debian system outputs the following error:

```./kovri: error while loading shared libraries: libboost_chrono.so.1.58.0: cannot open shared object file: No such file or directory```

I can include any others logs if needed

**Comment by anonimal** [comment]
Looks compiler memory related, 512mb cloud instance confirms. Working this out with @94b1 on IRC.

**Comment by anonimal** [comment]
Closed with #404
